app/Console/Services/Importer.php

```diff
@@ -423,4 +460,55 @@ protected function recalculateRanks()
     {
         /*$this->comment('--- RECALCULATING RANKS ---');*/
     }
+
+    /**
+     * Generate user's API Key and email them their new password
+     */
+    protected function setupUsers()
+    {
+        $allusers = User::all();
+        foreach($allusers as $user)
+        {
+            # Generate New User Password
+            # TODO: Increase Security?
+            $newpw = substr(md5(date('mdYhs')), 0, 10);
+
+            # Generate API Key
+            $apikey = Utils::generateApiKey();
+
+            # Update Info in DB
+            $user->password = bcrypt($newpw);
+            $user->api_key = $apikey;
+            $user->save();
+
+            # Send E-Mail to User with new login details
+            $email = new \App\Mail\NewLoginDetails($user, $newpw, 'New Login Details | '.config('app.name'));
```

Comment:
Don't email for now from here. We'll have to figure this bit out. Might be a separate option to do it, and we'll do it at the end, after the import has completed. Otherwise it slows it down considerably